pkg/kubelet/kubelet_pods.go

```diff
@@ -334,7 +334,14 @@ func ensureHostsFile(fileName string, hostIPs []string, hostName, hostDomainName
 		hostsFileContent = managedHostsFileContent(hostIPs, hostName, hostDomainName, hostAliases)
 	}
 
-	return ioutil.WriteFile(fileName, hostsFileContent, 0644)
+	oldMask, err := util.Umask(0)
+	if err != nil {
+		klog.Warning("failed to reset umask to zero, %v", err)
```

Comment:
Since this is a no-op on other platforms could we skip the error message (atleast on Windows)? Since this will happen for every pod on Windows we will have this error message which adds noise to the windows logs.